### PR TITLE
Styling MovieDetailsPage

### PR DESCRIPTION
**Pull request by sahedw**
Adapt the `onClick` function for the `BackButton` Component.
Create new `MovieDetailsFooter` Component to show a new fixed `footer` with buttons which add to lists instead of the `Navigation`.
Safe the preferred style (Dark Mode On/Off) in `local Storage`.
Adapt the function which displays the `watchProviders`to include three providers instead of two.
Adapt `jest` tests.
Add alternatives to icons and text if in dark mode (mainly color).

Use `Styled Components` to style:
Movie `Poster` and `Title `text.
`BackButton` Component.
`Actors` Component `image` and `name`.
Color for `heading` and `paragraph`.
Whole `Availability` section.

### CODE REVIEW

File: components/MovieDetail/index.js [hunk 1]
```diff
@@ -171,119 +219,80 @@ export default function MovieDetail({ movie }) {
...
+        <StyledMovieSubtitles>
+          {getPopularityDecimal(movieDetails?.vote_average)}/10 Rating
+        </StyledMovieSubtitles>
+        <StyledMovieSubtitles>
+          {getGenreFrom(movie)} • {movie.release_date.slice(0, 4)} •{" "}
+          {calculateRuntimeFrom(runtime)}
+        </StyledMovieSubtitles>
+      </StyledSectionQuickOverview>
+      <StyledSectionSynopsis>
+        {" "}
```
Comment (zamukaev) [inline components/MovieDetail/index.js:245]: do you need this here?

File: components/MovieDetail/index.js [hunk 2]
```diff
@@ -171,119 +219,80 @@ export default function MovieDetail({ movie }) {
...
-        <p>Flatrate: {`${showWatchProviders(streamingProviderFlatrate)}`}</p>
-      )}
-      {availabilityOption === "rent" && (
-        <p>Renting: {`${showWatchProviders(streamingProviderRent)}`}</p>
-      )}
+      <StyledSectionAvailability>
+        <h4>Availability:</h4>
+        {availabilityOption === "all" && (
+          <>
+            {" "}
```
Comment (zamukaev) [inline components/MovieDetail/index.js:269]: and here...?

File: components/MovieDetail/index.js [hunk 3]
```diff
@@ -171,119 +219,80 @@ export default function MovieDetail({ movie }) {
...
+            <StyledAvailabilityHeading>Flatrate</StyledAvailabilityHeading>
+            <StyledAvailabilityText>{`${showWatchProviders(
+              streamingProviderFlatrate
+            )}`}</StyledAvailabilityText>
+            <StyledAvailabilityHeading>Renting</StyledAvailabilityHeading>
+            <StyledAvailabilityText>
+              {" "}
```
Comment (zamukaev) [inline components/MovieDetail/index.js:276]: also here

**Review by zamukaev — APPROVED** [review]
look very good but

File: components/MovieDetail/index.js (hunk 2 above)
Comment (sahedw) [reply components/MovieDetail/index.js:269]: Same as above

**Review by zamukaev — COMMENTED** [review]
nice

**Comment by zamukaev** [comment]
look good

**Pull request merged**